### PR TITLE
Exclude publisher callback

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/5144
Resolves https://github.com/brave/brave-browser/issues/5319

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- enable rewards
- add some publishers to AC list
- go to rewards page and delete them from the list
- try to restore them as well

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc [hunk 1]
```diff
@@ -297,54 +297,53 @@ void BatPublishers::OnPublisherInfoSaved(
...
 
 void BatPublishers::OnSetPublisherExclude(
     ledger::PUBLISHER_EXCLUDE exclude,
     ledger::Result result,
-    ledger::PublisherInfoPtr publisher_info) {
+    ledger::PublisherInfoPtr publisher_info,
+    ledger::SetPublisherExcludeCallback callback) {
   if (result != ledger::Result::LEDGER_OK &&
       result != ledger::Result::NOT_FOUND) {
+    callback(ledger::Result::LEDGER_ERROR);
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc]: Seems like we should just return `result` here.

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc [hunk 2]
```diff
@@ -297,54 +297,53 @@ void BatPublishers::OnPublisherInfoSaved(
...
-  } else {
-    BLOG(ledger_, ledger::LogLevel::LOG_ERROR) <<
-      "Could not restore publishers.";
+void BatPublishers::OnRestorePublishers(
+    const ledger::Result result,
+    ledger::RestorePublishersCallback callback) {
+  if (result != ledger::Result::LEDGER_OK) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR)
+    << "Could not restore publishers.";
+    callback(ledger::Result::LEDGER_ERROR);
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc]: Seems like we should just return `result` here.

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 3]
```diff
@@ -750,11 +750,25 @@ base::PostTaskAndReplyWithResult(file_task_runner_.get(), FROM_HERE,
...
+  }
 
-  bat_ledger_->RestorePublishers();
+  for (auto& observer : observers_) {
+    observer.OnExcludedSitesChanged(
+      this, "-1", ledger::PUBLISHER_EXCLUDE::ALL);
+  }
+}
+
+void RewardsServiceImpl::RestorePublishersUI() {
```
Comment (tmancey) [inline components/brave_rewards/browser/rewards_service_impl.cc]: The naming convention used here is confusing as I am reading this in that we are restoring (or updating) the publishers UI, however speaking with @NejcZdovc this function is called from the UI.

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc (hunk 1 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc]: done

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc (hunk 2 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc]: done

**Comment by NejcZdovc** [comment]
CI passed last time https://staging.ci.brave.com/job/brave-core-build-pr/job/PR-2978/6/display/redirect
What I did in the last push is addressed comments. Tested it locally and it builds correctly.

**Pull request merged**